### DISCUSSION TITLE
in RULES how to subscribe to multiple values from sensors?

### DISCUSSION THREAD

**Discussion by zarfld** [arendst/Tasmota#20599] [Q&A] [answered]
Hello,
 i tcurrently tried to use the following rules:

`Rule1
ON mqtt#connected DO Backlog 
Subscribe SETROOMTEMP, stat/so_HzProberaum/SENSOR, AM2301.Temperature;
Subscribe SETVENTTEMP, stat/so_HzPub/SENSOR, AM2301.Temperature; 
Subscribe SETROOMHUM, stat/so_HzProberaum/SENSOR, AM2301.Humidity;
Subscribe SETROOMDEW, stat/so_HzProberaum/SENSOR, AM2301.DewPoint;
Subscribe SETVENTDEW, stat/so_HzPub/SENSOR, AM2301.DewPoint;
Subscribe SETROOMDEM, stat/so_HzProberaum/SENSOR, Thermostat1.TempTargetSet;
;event SETHUMMIN=50;event SETHUMMAX=65
ENDON
ON Event#SETROOMTEMP DO var1 %value% ENDON
ON Event#SETVENTTEMP DO var5 %value% ENDON
ON Event#SETROOMDEW DO var2 %value% ENDON
ON Event#SETVENTDEW DO var6 %value% ENDON
ON Event#SETROOMHUM DO var3 %value% ENDON
ON Event#SETROOMDEM DO var4 %value% ENDON
ON event#SETHUMMIN DO var7 %value% ENDON
ON event#SETHUMMAX DO var8 %value% ENDON

Rule2
ON Time#Minute|10 DO 
IF(%var6%<%var2% AND %var1%>%var4% AND %var1%>%var5% AND %var3%>%var7% ) POWER2 on 
ELSEIF (%var6%<%var2% AND %var1%<%var4% AND %var1%<%var5% AND %var3%>%var7% ) POWER2 on 
ELSEIF (%var6%>%var2% AND %var1%<%var4% AND %var1%<%var5% AND %var3%<%var7% ) POWER2 on 
ELSEIF (%var6%>%var2% AND %var1%>%var4% AND %var1%>%var5% AND %var3%<%var7% ) POWER2 on 
ELSEIF (%var6%<%var2% AND %var3%>%var8% ) POWER2 on 
ELSE POWER2 off 
ENDIF ENDON`

option SUPPORT_MQTT_EVENT, USE_EXPRESSION and SUPPORT_IF_STATEMENT are active.

it seems that the events which Subscribe to SENSORS are not getting triggered, i assume that ther might be a problem /conflict because i subscribe AM2301.Temperature, AM2301.DewPoint and Humidtity of the same SENSOR data. 

the output is the following - even after some time no events for Temp, Hum or Dewpoint are triggered - and vars are empty
>'21:09:41.659 RUL: MQTT#CONNECTED performs "Backlog  Subscribe SETROOMTEMP, stat/so_HzProberaum/SENSOR, AM2301.Temperature; Subscribe SETVENTTEMP, stat/so_HzPub/SENSOR, AM2301.Temperature;  Subscribe SETROOMHUM, stat/so_HzProberaum/SENSOR, AM2301.Humidity; Subscribe SETROOMDEW, stat/so_HzProberaum/SENSOR, AM2301.DewPoint; Subscribe SETVENTDEW, stat/so_HzPub/SENSOR, AM2301.DewPoint; Subscribe SETROOMDEM, stat/so_HzProberaum/SENSOR, Thermostat1.TempTargetSet; ;event SETHUMMIN=50;event SETHUMMAX=65"
21:09:42.013 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETROOMTEMP,stat/so_HzProberaum/SENSOR/#,AM2301.Temperature"}
21:09:42.266 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETVENTTEMP,stat/so_HzPub/SENSOR/#,AM2301.Temperature"}
21:09:42.516 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETROOMHUM,stat/so_HzProberaum/SENSOR/#,AM2301.Humidity"}
21:09:42.767 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETROOMDEW,stat/so_HzProberaum/SENSOR/#,AM2301.DewPoint"}
21:09:43.036 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETVENTDEW,stat/so_HzPub/SENSOR/#,AM2301.DewPoint"}
21:09:43.250 MQT: stat/so_VentKeller/RESULT = {"Subscribe":"SETROOMDEM,stat/so_HzProberaum/SENSOR/#,Thermostat1.TempTargetSet"}
21:09:43.497 MQT: stat/so_VentKeller/RESULT = {"Event":"Done"}
21:09:43.535 RUL: EVENT#SETHUMMIN performs "var7 50"
21:09:43.542 MQT: stat/so_VentKeller/RESULT = {"Var7":"50"}
21:09:43.760 MQT: stat/so_VentKeller/RESULT = {"Event":"Done"}
21:09:43.799 RUL: EVENT#SETHUMMAX performs "var8 65"
21:09:43.807 MQT: stat/so_VentKeller/RESULT = {"Var8":"65"}
21:09:44.366 QPC: Reset
'

**Comment by sfromis** [comment, marked as answer]
You're using the wrong topics. Telemetry data would not be published on `stat/so_HzProberaum/SENSOR` but instead `tele/so_HzProberaum/SENSOR`.

In general, do look in the console of the source device (or some MQTT subscription) to see exactly what you have available.

**Reply by zarfld** [reply]
yes thats right. I think i got that from https://tasmota.github.io/docs/MQTT/#subscribe where also "stat" was used for "SENSOR" but i could have checked on my own. thanks for pointing that out.

**Reply by sfromis** [reply]
Examples fixed :smile:
